contributions/api/utils/adminutils.py

```diff
@@ -54,54 +54,100 @@ def check_if_reviewer(login_id):
 
     return False
 
-def send_email(mail_to, subject, message):
+def establish_smtp_connection():
+    """
+    Method to establish SMTP connection
+    Args: None
+    Returns:
+        connection (obj): SMTP object
+        smtp_error (str): Error string
+        smtp_code (str): Error code
+    """
+    mail_from = cfg.SENDER_EMAIL
+    password = cfg.SENDER_EMAIL_PASSWORD
+    connection = smtplib.SMTP(host=cfg.SMTP_HOST, port=cfg.SMTP_PORT)
+    try:
+        connection.ehlo()
+    except smtplib.SMTPHeloError as ex:
+        return False, ex.smtp_error, ex.smtp_code
+    try:
+        connection.starttls()
+    except smtplib.SMTPConnectError as ex:
+        return False, ex.smtp_error, ex.smtp_code
+    try:
+        connection.login(mail_from, password)
+    except smtplib.SMTPAuthenticationError as ex:
+        return False, ex.smtp_error, ex.smtp_code
+    return connection, '', ''
+
+
+def test_smtp_connection(connection):
+    """
+    Method to test the SMTP connection
+    Args:
+        connection (obj): SMTP object
+    Returns:
+        (bool): True if connection is open, False if not
+    """
+    try:
+        status = connection.noop()[0]
+    except smtplib.SMTPServerDisconnected as ex:
+        status = -1
+    return True if status == 250 else False
+
+
+def quit_smtp_connection(connection):
```

Comment:
This needs to be called from somewhere right? I think https://flask.palletsprojects.com/en/2.1.x/api/#flask.Flask.teardown_appcontext is the appropriate method to register methods to be called when the app quits. Please see the example here: https://flask.palletsprojects.com/en/2.1.x/tutorial/database/#register-with-the-application

Reply:
Will work on establishing and closing the connection in contributions_rest_service.